### PR TITLE
Fix missing react in create-react-class

### PR DESCRIPTION
**Pull request by mondwan**
refs #9689

### CODE REVIEW

**Comment by gaearon** [comment]
Note: we'll need to figure out how to recompile this file for `.min.js` to also have this change.

**Comment by flarnie** [comment]
I believe CI is failing because this was made against 15-stable, and CI is still failing there.

**Comment by flarnie** [comment]
This is close to being mergable - I think we are a bit worried about how this change could affect the other non-AMD uses. What if we model this after the way that `react-dom` works? (credit to @gaearon for thinking of this):
18697850_420365161667842_787782393_o
Then I think we could skip the `var React = require('react');` in the body of the addon.

**Comment by flarnie** [comment]
@mondwan I'm going to pull this down and do some testing, and might add a commit to get it working for all builds.

**Comment by mondwan** [comment]
Great. Let me know if there are anything I can help.

2017年6月8日 07:27，"Flarnie Marchan" <notifications@github.com>寫道：

@mondwan <https://github.com/mondwan> I'm going to pull this down and do
some testing, and might add a commit to get it working for all builds.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/facebook/react/pull/9761#issuecomment-306954039>, or mute
the thread

**Comment by flarnie** [comment]
Almost have this working - but in general we are going to need a more maintainable way of updating add-ons going forward.

**Comment by flarnie** [comment]
Pushed some commits on top of the original fix - here is an explanation:
Here is the deal:
 - @mondwan's original fix does fix the AMD build, but breaks the
   'global JS' build.
 - My modification makes it work with both AMD and the 'global JS'
   build.
 - @mondwan's fix seems to have fixed the CommonJS build too, and I
   maintained that fix with my modification.

```
                Does the 'create-react-class' UMD build work?

                 Before       After         After
               + @mondwan's + @mondwan's +  @flarnie's
  Build System | fix        | fix        |  modification
+---------------------------------------------------------+
               |            |            |
  Global JS    | :D Success | X Fail     | :D Success
               |            |            |
+---------------------------------------------------------+
               |            |            |
  AMD          | X Fail     | :D Success | :D Success
               |            |            |
+---------------------------------------------------------+
               |            |            |
  Common JS    | X Fail     | :D Success | :D Success
               |            |            |
               +            +            +

```

I tested the fix manually using the fixtures - can explain if needed.

@gaearon would be interested in your ideas about this. Looks like we could take this approach to fix the other add-ons. I think we could land this, then update the minified 'create-react-class' and fix the other add-ons in a separate PR.

**Comment by mondwan** [comment]
Hello, it seems this line should also be removed?

https://github.com/mondwan/react/blob/15-stable/addons/create-react-class/create-react-class.js#L20

File: addons/create-react-class/create-react-class.js [hunk 1]
```diff
@@ -17,9 +17,10 @@
     if (typeof g.React === "undefined") {
       throw Error('React module should be required before createClass');
     }
-    g.createReactClass = f();
+    console.log('TEST');
```
Comment (flarnie) [inline addons/create-react-class/create-react-class.js]: will remove this

**Comment by flarnie** [comment]
CI is going to fail because it is always currently failing on the 15-stable branch. That will be fixed by the 15.6 release.

Here are my thoughts - 
1) Merge this
2) cherry-pick it into 15.6-dev
3) fix the similar issue in `create-fragment` and `react-linked-input` (see https://github.com/facebook/react/issues/9765) on 15.6-dev
4) merge those fixes (hopefully this weekend)
5) the fixes will be included in the 15.6 release on Tuesday, June 13th.

**Comment by mondwan** [comment]
Hey @flarnie , I have a question outside for this ticket's scope.

May I know how to pop out below contents in github?

```
flarnie reviewed 8 days ago View changes
Show outdated addons/create-react-class/create-react-class.js
```

It is useful for highlighting that particular line with a review icon. How to click such thing from the UI panel in github?

**Pull request merged**